web3/utils/validation.py

```diff
@@ -51,6 +52,16 @@ def validate_abi_value(abi_type, value):
     Note: abi_type 'bytes' must either be python3 'bytes' object or ''
     """
     if is_array_type(abi_type) and is_list_like(value):
+        # validate length
+        specified_length = length_of_array_type(abi_type)
+        if specified_length and (specified_length != len(value)):
```

Comment:
Should this be `if specified_length is not None and ...` to account for the edge case of zero length arrays?  or are those even a thing?